### DISCUSSION TITLE
How set API rule to check login user is of specific Auth Model

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2080] [Q&A] [answered]
Since different auth collection can be created, i would like to set API rule of another collection only user for specifc collection able to edit them.  Through @request.auth.id != ""  i can able to make sure whether user is authenticated or not but how to check collection he belong to

**Comment by ganigeorgiev** [comment, marked as answer]
You can check `@request.auth.collectionId` or `@request.auth.collectionName`.

**Reply by iamdineshkumar** [reply]
whether collectionId & collectionName are same? in case difference where can i find collectionId?

**Reply by ganigeorgiev** [reply]
`collectionId` is the id of the collection (random 15chars string; once created can be changed only programmatically). You can find the collection id in the Admin UI URL bar, in the user model response, etc.

`collectionName` is the name of the collection, eg. "users" (this value changes if you rename your collection).

**Reply by iamdineshkumar** [reply]
In Auth collection, I have a field as userType in case of its value is Supervisor, I need to allow user to edit collection records or should restrict. How to achieve with API rule?

**Reply by ganigeorgiev** [reply]
@iamdineshkumar 

It depends on your collection structure, what are your specific requirements and what exactly do you mean by _"edit collection records"_.

If you want users to be able to change only their own account fields with exception of the `userType` (make sure that it is nonempty!), then you can use the following rules set:

---

**Create API rule**

_Ensures that users can't create on their own "supervisor" accounts._

```js
@request.data.userType = "regular"
```

---

**Update API rule**

_Ensures that only the supervisors OR the user owner can modify the account._
_"Supervisor" users are allowed to change all fields of all users, while the regular users can modify only their own account, disallowing `userType` change._

```js
@request.auth.userType = "supervisor" ||
(
  id = @request.auth.id &&
  (@request.data.userType:isset = false || @request.data.userType = "regular")
)
```

---

**Manage API rule** (optional)

_Allows user with type "supervisor" to change directly the user `verified` status, email, password, etc. without confirimation._

```js
@request.auth.userType = "supervisor"
```

---

Note that I haven't tested any of the above rules and you may have to adjust them based on your use case.

**Reply by iamdineshkumar** [reply]
Thanks, I will check those.... For confirmation. We can able to access all fields available in auth collection of request user by 
```
@request.auth.[fieldName]
```

**Reply by ganigeorgiev** [reply]
@iamdineshkumar Yes, you can access the current authenticated record fields using dot-notation. Relations are also supported:

```js
@request.auth.someMainRelField.subRelA.subRelB.status = "active"
```
